ir/expression.def

```diff
@@ -269,7 +269,13 @@ class Mux : Operation_Ternary {
 
 class DefaultExpression : Expression {}
 
-class This : Expression {}  // experimental
+// Two different This should not be equal.
+// That's why we use a hidden id field to distinguish them.
+class This : Expression {
+    int id = nextId++;
+ private:
+    static int nextId;
+}  // experimental
```

Comment:
I am pretty sure it would work correctly for all the passes we currently have - in the reference map they will both point to the same declaration.